### PR TITLE
added components

### PR DESCRIPTION
**Pull request by kv-yan**
Added action bar for auth flow , buton , text field

### CODE REVIEW

File: app/src/main/java/com/simply/birthdayapp/commonpresentation/components/button/AuthButton.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.simply.birthdayapp.commonpresentation.theme.ButtonDisableColor
+import com.simply.birthdayapp.commonpresentation.theme.ButtonEnableColor
+import com.simply.birthdayapp.commonpresentation.theme.ButtonPink
+import com.simply.birthdayapp.commonpresentation.theme.ButtonPurple
+import com.simply.birthdayapp.commonpresentation.theme.ButtonTextStyle
+
+@Composable
+fun AuthedButton(
```
Comment (pmuradian) [inline app/src/main/java/com/simply/birthdayapp/commonpresentation/components/button/AuthButton.kt:18]: File is called AuthButton, function is called AuthedButton

File: app/src/main/java/com/simply/birthdayapp/commonpresentation/components/textfield/InputTextField.kt [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                    verticalAlignment = Alignment.CenterVertically,
+                    modifier = Modifier
+                        .background(
+                            color = if (isHasError) TextFieldErrorContainerColor else TextFieldContainerColor,
+                            shape = TextFieldShape
+                        )
+                        .padding(horizontal = 12.dp, vertical = 0.dp)
+                        .fillMaxWidth()
+                ) {
+                    if (leadingIcon != null) {
```
Comment (pmuradian) [inline app/src/main/java/com/simply/birthdayapp/commonpresentation/components/textfield/InputTextField.kt:77]: leadingIcon?.let notation could be used

File: app/src/main/java/com/simply/birthdayapp/commonpresentation/components/textfield/InputTextField.kt [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    isHasError: Boolean = false,
+    isPassword: Boolean = false,
+    leadingIcon: @Composable (() -> Unit)? = null,
+    trailingIcon: @Composable (() -> Unit)? = null,
+) {
+    var passwordVisible by remember { mutableStateOf(false) }
+
+    Box(modifier = modifier) {
+        BasicTextField(
+            value = value.value,
```
Comment (pmuradian) [inline app/src/main/java/com/simply/birthdayapp/commonpresentation/components/textfield/InputTextField.kt:58]: pick a better name for value argument, value.value does not look nice

File: app/src/main/java/com/simply/birthdayapp/commonpresentation/components/textfield/InputTextField.kt [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            onValueChange = { value.value = it },
+            singleLine = true,
+            textStyle = TextFieldTextStyle.copy(textAlign = TextAlign.Start),
+            modifier = Modifier
+                .fillMaxWidth()
+                .then(modifier),
```
Comment (pmuradian) [inline app/src/main/java/com/simply/birthdayapp/commonpresentation/components/textfield/InputTextField.kt:64]: looks suspicious, why not write modifier.fillMaxWidth ?

File: app/src/main/java/com/simply/birthdayapp/commonpresentation/components/textfield/InputTextField.kt (hunk 2 above)
Comment (kv-yan) [reply app/src/main/java/com/simply/birthdayapp/commonpresentation/components/textfield/InputTextField.kt:77]: leadingIcon?.invoke()

can i use invoke with "elvis"this ?

File: app/src/main/java/com/simply/birthdayapp/commonpresentation/components/textfield/InputTextField.kt (hunk 4 above)
Comment (kv-yan) [reply app/src/main/java/com/simply/birthdayapp/commonpresentation/components/textfield/InputTextField.kt:64]: I didn't notice, maybe it did auto-completed 
removed

**Pull request merged**